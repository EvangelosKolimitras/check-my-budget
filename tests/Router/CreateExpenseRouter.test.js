import React from "react"
import toJson from "enzyme-to-json"
import { Add } from "../../src/scripts/js/Router/CreateExpenseRouter"
import { shallow, configure } from "enzyme"
import Adapter from "enzyme-adapter-react-16"
configure({ adapter: new Adapter() })

describe("Routers:", () => {
    describe("the CreateExpenseRouter router,", () => {
        it("should render CreateExpenseRouter correctly", () => {
            const onSubmit = jest.fn()
            const history = {
                push: jest.fn()
            }
            const wrapper = shallow(<Add onSubmit={onSubmit} history={history} />)
            expect(toJson(wrapper)).toMatchSnapshot()
        })
    })
})
