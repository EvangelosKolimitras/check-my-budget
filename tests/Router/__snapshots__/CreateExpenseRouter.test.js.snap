// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routers: the CreateExpenseRouter router, should render CreateExpenseRouter correctly 1`] = `
<div>
  <h1>
    Add an expense
  </h1>
  <ExpenseForm
    onSubmit={[Function]}
  />
</div>
`;
